python/lsst/ts/watcher/rules/mt_hexapod_high_current.py

```diff
@@ -0,0 +1,121 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MTHexapodHighCurrent"]
+
+
+import logging
+import types
+import typing
+
+from lsst.ts import salobj, watcher
+from lsst.ts.xml.enums.MTHexapod import SalIndex
+from lsst.ts.xml.enums.Watcher import AlarmSeverity
+
+from ..base_rule import AlarmSeverityReasonType, NoneNoReason
+from ..remote_info import RemoteInfo
+
+
+class MTHexapodHighCurrent(watcher.BaseRule):
+    """Monitor the high motor current event of the MT hexapod.
+
+    Parameters
+    ----------
+    config : `types.SimpleNamespace`
+        Rule configuration, as validated by the schema.
+    log : `logging.Logger` or None, optional
+        Parent logger. (the default is None)
+    """
+
+    def __init__(
+        self, config: types.SimpleNamespace, log: logging.Logger | None = None
+    ):
+
+        remote_name = "MTHexapod"
+        remote_info_camera = RemoteInfo(
+            remote_name,
+            1,
+            callback_names=["evt_highCurrent"],
+        )
+        remote_info_m2 = RemoteInfo(
+            remote_name,
+            2,
+            callback_names=["evt_highCurrent"],
+        )
+
+        super().__init__(
+            config,
+            f"MTHexapodHighCurrent.{remote_name}",
+            [remote_info_camera, remote_info_m2],
+            log=log,
+        )
+
+    @classmethod
+    def get_schema(cls):
+        # No schema necessary for this rule.
+        return None
+
+    def compute_alarm_severity(
+        self, data: salobj.BaseMsgType, **kwargs: dict[str, typing.Any]
+    ) -> AlarmSeverityReasonType:
+        """Compute and set alarm severity and reason.
+
+        Parameters
+        ----------
+        data : `salobj.BaseMsgType`
+              Message from the topic described by topic_callback.
+        **kwargs : `dict` [`str`, `typing.Any`]
+            Keyword arguments. If triggered by `TopicCallback` calling
+            `update_alarm_severity`, the arguments will be as follows:
+
+            * topic_callback : `TopicCallback`
+              Topic callback wrapper.
+
+        Returns
+        -------
+        None, if no change or unknown, or a tuple of two values:
+
+        severity: `lsst.ts.xml.enums.Watcher.AlarmSeverity`
+            The new alarm severity.
+        reason : `str`
+            Detailed reason for the severity, e.g. a string describing
+            what value is out of range, and what the range is.
+            If ``severity`` is ``NONE`` then this value is ignored (but still
+            required) and the old reason is retained until the alarm is reset
+            to ``nominal`` state.
+
+        Notes
+        -----
+        You may return `NoneNoReason` if the alarm state is ``NONE``.
+        """
+
+        sal_index = SalIndex(data.salIndex)
```

Comment:
yeah, I am not really sure I agree with this implementation. Why not let the alarm read the current and compute the alarm severity here? I don't think relying on the CSC to send this is a good idea.

Reply:
Even in the normal using, the hexapod can have the current values to be higher than the threshold. This is normal because the Copley controllers needs to have a significant torque to overcome the high friction as a reality of design flaw we have. My original plan was to use the hexapod CSC to work on this logic because it would be easier. But we could try to put the complexity in the watcher as well although it would need to base on multiple events and telemetry to judge this. Thanks!